src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -170,5 +177,95 @@ public RestClient FindRestClient(OperationGroup operationGroup)
             EnsureHeaderModels().TryGetValue(operation, out var model);
             return model;
         }
+
+        private void DecorateOperationGroup()
+        {
+            foreach (var operations in _codeModel.OperationGroups)
+            {
+                operations.ProviderName = _context.Configuration.OperationGroupMapping.ContainsKey(operations.Key) ? _context.Configuration.OperationGroupMapping[operations.Key] : ConstructOperationProviderName(operations);
+            }
+        }
+
+        private string ConstructOperationProviderName(OperationGroup operations)
+        {
+
+            string? providerName = "";
+            var request = GetBestMethod(operations);
+            bool adding = false;
+            if (request != null)
+            {
+                foreach (var segment in GetPathSegments(request.Path))
+                {
+                    var asSplit = segment.Split('/', StringSplitOptions.RemoveEmptyEntries);
+                    if (asSplit?.Length > 1 && asSplit.First().Equals("providers"))
+                    {
+                        adding = true;
+                        providerName = segment.Substring("providers".Length + 2).TrimEnd('/');
+                    }
+                    else if (adding)
+                    {
+                        providerName += segment.TrimEnd('/');
+                    }
+                }
+            }
+            return providerName.TrimEnd('/');
+        }
+        private static List<string> GetPathSegments(string httpRequestUri)
+        {
+            List<string> seg = new List<string>();
+            string canidate = "";
+
+            foreach (var ch in httpRequestUri)
+            {
+                if (ch == '{')
+                {
+                    if (canidate != "" && canidate != "/")
+                    {
+                        seg.Add(canidate);
+                    }
+                    canidate = "";
+                }
+                else if (ch == '}')
+                {
+                    canidate = "";
+                }
+                else
+                {
+                    canidate += ch;
+                }
+            }
+            if (canidate != "" && canidate != "/")
+            {
+                seg.Add(canidate);
+            }
+            return seg;
```

Comment:
doesn't appear this is getting path segments, it appears its getting all reference portions of the path

Can we update the name?